<h1>PALMER PENGUINS DATA ANALYTICS PROJECT</h1>
<hr>
<h3>Analyst : Lee Ann Tabanao</h3>

In [167]:
import numpy as np
import pandas as pd
import warnings 
warnings.filterwarnings('ignore')

<h2>Covert dataset(s) into dataframe</h2>
<h3>Reguirements</h3>
<ul>
    <li>
        Name of Dataframe
    </li>
    <li>
        File Path
    </li>
</ul>

In [2]:
penguins = pd.read_csv('datasets//penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h2>Descriptive Analytics</h2>

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

<h2>Checking for null values</h2>

In [5]:
penguins.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

In [6]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [7]:
penguins['island']

0      Torgersen
1      Torgersen
2      Torgersen
3      Torgersen
4      Torgersen
         ...    
339        Dream
340        Dream
341        Dream
342        Dream
343        Dream
Name: island, Length: 344, dtype: object

In [8]:
penguins['island'] != 'togersen'

0      True
1      True
2      True
3      True
4      True
       ... 
339    True
340    True
341    True
342    True
343    True
Name: island, Length: 344, dtype: bool

In [9]:
penguins[penguins['island'] != 'togersen']

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

In [10]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [11]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [12]:
penguins[penguins['species'] == 'chinstrap']

Empty DataFrame
Columns: [species, island, bill_length_mm, bill_depth_mm, flipper_length_mm, body_mass_g, sex, year]
Index: []

In [13]:
penguins[penguins['species'] == 'Gentoo']

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [14]:
penguins[penguins['species'] == 'Adelie'] 

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [15]:
penguins[penguins['body_mass_g'] >= 4000]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7       Adelie  Torgersen            39.2           19.6              195.0   
9       Adelie  Torgersen            42.0           20.2              190.0   
14      Adelie  Torgersen            34.6           21.1              198.0   
17      Adelie  Torgersen            42.5           20.7              197.0   
19      Adelie  Torgersen            46.0           21.5              194.0   
..         ...        ...             ...            ...                ...   
323  Chinstrap      Dream            49.0           19.6              212.0   
329  Chinstrap      Dream            50.7           19.7              203.0   
333  Chinstrap      Dream            49.3           19.9              203.0   
339  Chinstrap      Dream            55.8           19.8              207.0   
342  Chinstrap      Dream            50.8           19.0              210.0   

     body_mass_g   sex  year  
7         4675.0  male  2007  
9         4250.0   NaN  2007  
14        4400.0  male  2007  
17        4500.0  male  2007  
19        4200.0  male  2007  
..           ...   ...   ...  
323       4300.0  male  2009  
329       4050.0  male  2009  
333       4050.0  male  2009  
339       4000.0  male  2009  
342       4100.0  male  2009  

[177 rows x 8 columns]

In [16]:
Gentoo_mass = penguins[(penguins['species'] == 'Gentoo') & (penguins['body_mass_g'] >= 4000)]

In [17]:
Gentoo_mass

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
270  Gentoo  Biscoe            47.2           13.7              214.0   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
270       4925.0  female  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[122 rows x 8 columns]

<h2>Filling up null values</h2>

In [18]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [19]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [20]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [21]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [22]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [23]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [24]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       2
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [25]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [26]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [27]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [28]:
penguins['bill_depth_mm'].mean()

np.float64(17.151169590643274)

In [29]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [30]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    2
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [31]:
penguins['flipper_length_mm'].mean()

np.float64(200.91520467836258)

In [32]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [33]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h2>Changing the data type</h2>

In [34]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [35]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')
penguins['year'] = penguins['year'].astype('category')

In [36]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


In [37]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      344.000000     344.000000         344.000000   344.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.443643       1.969027          14.020657   799.613058
min         32.100000      13.100000         172.000000  2700.000000
25%         39.275000      15.600000         190.000000  3550.000000
50%         44.250000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

In [38]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  
0    3750.000000    male  2007  
1    3800.000000  female  2007  
2    3250.000000  female  2007  
3    4201.754386    male  2007  
4    3450.000000  female  2007  
..           ...     ...   ...  
339  4000.000000    male  2009  
340  3400.000000  female  2009  
341  3775.000000    male  2009  
342  4100.000000    male  2009  
343  3775.000000  female  2009  

[344 rows x 8 columns]

In [39]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] /1000,1)

In [40]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

<hr>
<h1>*Explanatory Data Analytics*</h1>

<hr>
<h2>Question #1</h2>
<h4>Which species has the highest number of females?</h4>

<ul>
    <li>
        <h3>Gentoo</h3>
    </li>
</ul>

In [41]:
penguins[penguins['species'] == 'Gentoo'].value_counts('sex')

sex
male      66
female    58
Name: count, dtype: int64

<ul>
    <li>
        <h3>Adelie</h3>
    </li>
</ul>

In [42]:
penguins[penguins['species'] == 'Adelie'].value_counts('sex')

sex
male      79
female    73
Name: count, dtype: int64

<ul>
    <li>
        <h3>Chinstrap</h3>
    </li>
</ul>

In [43]:
penguins[penguins['species'] == 'Chinstrap'].value_counts('sex')

sex
female    34
male      34
Name: count, dtype: int64

<h2>Insight #1</h2>
<h4>In this evaluation, we observe that the Adelie species has the highest number of females.</h4>

<hr>
<h2>Question #2</h2>
<h4>Which penguin species has the longest and the shortest flipper length?</h4>

In [44]:
penguins['flipper_length_mm'].max()

np.float64(231.0)

<ul>
    <li>
        <h3>Longest</h3>
    </li>
</ul>

In [45]:
penguins[penguins['flipper_length_mm'] == 231.0]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
215  Gentoo  Biscoe            54.3           15.7              231.0   

     body_mass_g   sex  year  mass_kg  
215       5650.0  male  2008      5.6

In [46]:
penguins['flipper_length_mm'].min()

np.float64(172.0)

<ul>
    <li>
        <h3>Shortest</h3>
    </li>
</ul>

In [47]:
penguins[penguins['flipper_length_mm'] == 172.0]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
28  Adelie  Biscoe            37.9           18.6              172.0   

    body_mass_g     sex  year  mass_kg  
28       3150.0  female  2007      3.2

<h2>Insight #2</h2>
<h4>In the Gentoo species, we observe that they have the longest flippers, while the Adelie species have the shortest.</h4>

<hr>
<h2>Question #3</h2>
<h4>Which year has the highest number of penguin observations?</h4>

In [48]:
penguins['year'].value_counts()

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

<h2>Insight #3</h2>
<h4>In the year 2009, the highest number of penguin observations was recorded.</h4>

<hr>
<h2>Question #4</h2>
<h4>Which island has the most penguin observations?</h4>

In [49]:
penguins['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h2>Insight #4</h2>
<h4>After analyzing the data, it is clear that Biscoe Island has the highest number of penguin observations.</h4>

<hr>
<h2>Question #5</h2>
<h4>How many penguins weigh over 5000 grams?</h4>

In [50]:
penguins[penguins['body_mass_g'] >= 5000].value_counts('body_mass_g')

body_mass_g
5000.0    6
5550.0    6
5700.0    5
5500.0    5
5400.0    5
5200.0    4
5300.0    4
5250.0    3
5100.0    3
5050.0    3
5850.0    3
5650.0    3
5350.0    3
5150.0    2
6000.0    2
5600.0    2
5800.0    2
5950.0    2
5450.0    1
5750.0    1
6050.0    1
6300.0    1
Name: count, dtype: int64

In [51]:
peguins_heavy = penguins[penguins['body_mass_g'] >= 5000].value_counts('body_mass_g')

In [52]:
peguins_heavy.sum()

np.int64(67)

<h2>Insight #5</h2>
<h4>After analyzing the data, it shows that there are 67 penguins that weigh over 5000 grams.</h4>

<hr>
<h2>Question #6</h2>
<h4>What is the average bill length of all penguins?</h4>

In [53]:
penguins['bill_length_mm'].mean()

np.float64(43.92192982456141)

<h2>Insight #6</h2>
<h4>The average bill length of all penguins in the dataset is about 43.92 mm</h4>

<hr>
<h2>Question #7</h2>
<h4>What is the lightest penguin recorded and on which island was it found?</h4>

In [54]:
penguins['body_mass_g'].min()

np.float64(2700.0)

In [55]:
penguins[penguins['body_mass_g'] == 2700.0]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
314  Chinstrap  Dream            46.9           16.6              192.0   

     body_mass_g     sex  year  mass_kg  
314       2700.0  female  2008      2.7

<h2>Insight #7</h2>
<h4>The lightest penguin recorded weighed 2700g and was found on Dream Island.</h4>


<hr>
<h2>Question #8</h2>
<h4>What are the minimum and maximum bill lengths recorded?</h4>

In [69]:
min_bill = penguins.groupby(['species'])['bill_length_mm'].max()

In [70]:
min_bill

species
Adelie       46.0
Chinstrap    58.0
Gentoo       59.6
Name: bill_length_mm, dtype: float64

In [71]:
max_bill = penguins.groupby(['species'])['bill_length_mm'].min()

In [73]:
max_bill

species
Adelie       32.1
Chinstrap    40.9
Gentoo       40.9
Name: bill_length_mm, dtype: float64

<h2>Insight #8</h2>
<h4>The smallest bill length recorded is 32.1 mm in Adelie penguins, while the largest is 59.6 mm in Gentoo penguins, showing considerable variation in bill size across species.</h4>

<hr>
<h2>Question #9</h2>
<h4>Which species shows the largest difference in average body mass between sexes?</h4>

In [104]:
mass = penguins.groupby(['species', 'sex'])['body_mass_g'].max()

mass

In [106]:
mass_min = penguins.groupby(['species', 'sex'])['body_mass_g'].min()

In [107]:
mass_min

species    sex   
Adelie     female    2850.0
           male      2975.0
Chinstrap  female    2700.0
           male      3250.0
Gentoo     female    3950.0
           male      4100.0
Name: body_mass_g, dtype: float64

<h2>Insight #9</h2>
<h4>Among the penguin species, Gentoo penguins show the largest difference in average body mass between males and females, with males being significantly heavier than females.</h4>


<hr>
<h2>Question #10</h2>
<h4> Is there a correlation between bill length and body mass?</h4>

In [122]:
relation = penguins['bill_length_mm'].corr(penguins['body_mass_g'])

In [123]:
relation

np.float64(0.5951098244376304)

<h2>Insight #10</h2>
<h4>There is a moderately strong positive correlation between bill length and body mass in penguins, meaning that penguins with longer bills tend to have higher body mass.</h4>


<hr>
<h2>Question #11</h2>
<h4>Did the average bill length change from 2007 to 2009?</h4>

In [137]:
av_length = penguins.groupby(['species','year'])['bill_length_mm'].mean()

In [138]:
av_length

species    year
Adelie     2007    38.926439
           2008    38.560000
           2009    38.982692
Chinstrap  2007    48.723077
           2008    48.700000
           2009    49.054167
Gentoo     2007    47.014706
           2008    46.936957
           2009    48.395953
Name: bill_length_mm, dtype: float64

<h2>Insight #11</h2>
<h4>Yes, the average bill length changed from 2007 to 2009: Adelie penguins stayed almost the same, Chinstrap penguins showed a slight increase, while Gentoo penguins showed a small decrease.</h4>

<hr>
<h2>Question #12</h2>
<h4>Is there a noticeable bill size difference between penguins?</h4>

In [218]:
from scipy.stats import f_oneway

In [235]:
penguins[penguins['species'] == 'Adelie']['bill_length_mm']

0      39.10000
1      39.50000
2      40.30000
3      43.92193
4      36.70000
         ...   
147    36.60000
148    36.00000
149    37.80000
150    36.00000
151    41.50000
Name: bill_length_mm, Length: 152, dtype: float64

In [236]:
penguins[penguins['species'] == 'Chinstrap']['bill_length_mm']

276    46.5
277    50.0
278    51.3
279    45.4
280    52.7
       ... 
339    55.8
340    43.5
341    49.6
342    50.8
343    50.2
Name: bill_length_mm, Length: 68, dtype: float64

In [237]:
penguins[penguins['species'] == 'Gentoo']['bill_length_mm']

152    46.10000
153    50.00000
154    48.70000
155    50.00000
156    47.60000
         ...   
271    43.92193
272    46.80000
273    50.40000
274    45.20000
275    49.90000
Name: bill_length_mm, Length: 124, dtype: float64

In [238]:
adelie_length = penguins[penguins['species'] == 'Adelie']['bill_length_mm']
chinstrap_length = penguins[penguins['species'] == 'Chinstrap']['bill_length_mm']
gentoo_length = penguins[penguins['species'] == 'Gentoo']['bill_length_mm']

In [239]:
f_statistic, p_value = f_oneway(adelie_length, chinstrap_length, gentoo_length)

In [240]:
p_value

np.float64(7.233266188149441e-91)

<h2>Insight #12</h2>
<h4>The very small p-value 7.23e-91 shows there is a highly significant difference in bill size among Adelie, Chinstrap, and Gentoo penguins.</h4>

<hr>
<h2>Question #13</h2>
<h4> Do male penguins generally have longer flippers than females?</h4>

In [253]:
penguins[penguins['sex'] == 'male']['flipper_length_mm'].max()

np.float64(231.0)

In [254]:
penguins[penguins['sex'] == 'female']['flipper_length_mm'].max()

np.float64(222.0)

In [255]:
male_bill = penguins[penguins['sex'] == 'male']['flipper_length_mm']
female_bill = penguins[penguins['sex'] == 'female']['flipper_length_mm']

In [256]:
f_v, p_value = f_oneway(male_bill, female_bill)

In [257]:
p_value

np.float64(4.88065474065626e-06)

<h2>Insight #13</h2>
<h4>male penguins generally have longer flippers than females, and the very small p-value 4.9e-06 confirms this difference is statistically significant.</h4>
<hr>